src/perf_tool_backend.ml

```diff
@@ -46,6 +46,19 @@ module Record_opts = struct
   ;;
 end
 
+let write_perf_dlfilter () =
+  let error = Deferred.Or_error.error_string "Unable to write [perf_dlfilter.so]." in
+  try
+    match Perf_dlfilter.read "perf_dlfilter.so" with
```

Comment:
Let's put this in the `-working-dir`.

Reply:
Do we want to leave this around after magic-trace runs though? Do you think it should delete the shared object at the end of execution?